core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -391,6 +395,18 @@ void connect(SocketAddress remoteAddress, SessionProtocol desiredProtocol,
                 final Channel channel = registerFuture.channel();
                 configureProxy(channel, poolKey.proxyConfig, desiredProtocol);
 
+                if (desiredProtocol.isTls() && timingsBuilder != null) {
+                    channel.attr(TIMINGS_BUILDER_KEY).set(timingsBuilder);
+
+                    // If TCP fast open is enabled,
+                    // Should start the TLS handshake timer after` before connecting the TCP connection.
```

Comment:
Setting the channel option doesn't necessarily mean that TFO is being used.
The OS has to support TFO, and the cookie has to be already cached.

https://github.com/netty/netty/blob/2e413d6c73aac00d4541bab119c7a2c330bf8235/transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannel.java#L126-L136

I'm concerned that this will give a wrong impression that enabling TFO is slower than the disabling it.

Rather, I wonder if just always measuring between `channelActive ~ handshake completion event` is 1) simpler 2) and yields more consistent results (if TFO is enabled, I guess the handshake will be completed faster than disabling it).

What do you think other maintainers? @line/dx

Reply:
No objections. 😆

Reply:
Then, can I change it to measure TLS handshake timing only with `channelActive ~ hanshake completion` without considering a separate processing for TFO option???

Reply:
That is correct. 👍